### PR TITLE
improve tooltip

### PR DESCRIPTION
**Pull request by Mahdi-zarei**

### CODE REVIEW

File: src/visualization.ts [hunk 1]
```diff
@@ -95,24 +97,40 @@ export class GraphVisualizer {
...
-                    // @ts-ignore
-                    const tip = d3Tip()
-                        .attr('class', 'd3-tip')
-                        .offset([-10, 0])
-                        .html((event: any, d: Node) => {
-                            return d.description;
-                        })
-                    d3.select(this.container).call(tip);
+                    let tip = new Tooltip();
+                    document.body.appendChild(tip.getNode());
```
Comment (Shayan-P) [inline src/visualization.ts:101]: add to body or add to this.container?

File: src/visualization.ts [hunk 2]
```diff
@@ -95,24 +97,40 @@ export class GraphVisualizer {
...
+                        linger = false;
+                        lingerTimeout = setTimeout(this.hideTask, 800)
+                    });
+
                     nodeEnter.filter((d: Node) => {
                         return d.description !== undefined && d.description.length > 0;
                     }).on('mouseover', function (...args) {
+                        keep = true;
+                        clearTimeout(showTimeout)
+                        tip.setData(...args, that.config.circleRadiusUnit);
```
Comment (Shayan-P) [inline src/visualization.ts:117]: why do we need this?

File: src/visualization.ts (hunk 1 above)
Comment (Mahdi-zarei) [reply src/visualization.ts:101]: since the position is absolute, I guess anywhere in the DOM is fine.

File: src/visualization.ts (hunk 2 above)
Comment (Mahdi-zarei) [reply src/visualization.ts:117]: resolved in [remove redundant code](https://github.com/Shayan-P/Shayan-P.github.io/pull/1/commits/dc7f7d314af40ddba7af6722497771ade6564a1f)

**Pull request merged**